contrib/automation_tests/test_cases/connection_window.py

```diff
@@ -95,13 +96,93 @@ def _execute(self):
         self.expect_true(instance_list_overlay.is_visible(), "Instance overlay is visible")
         logging.info('Found InstanceListOverlay and its visible')
 
-        window = self.suite.top_window()
-        window.InstanceList.DataItem0.wait('exists', timeout=100)
-        self.expect_true(instance_list_overlay.rectangle().width() == 0,
-                         "InstanceOverlay has width() 0 (is not shown)")
+        wait_for_condition(
+            lambda: self.find_control('Group', 'InstanceListOverlay', raise_on_failure=False) is
+            None, 100)
         logging.info('Loading done, overlay is hidden')
-        logging.info('Found %s rows in the instance list', instance_list.item_count())
-        self.expect_true(instance_list.item_count() >= 1, 'Found at least one instance')
+        self.expect_true(get_number_of_instances_in_list(self) >= 1, 'Found at least one instance')
+
+
+class SelectProject(E2ETestCase):
+    """
+    Expand the project selection combo box and click the entry `project_name`. Also waits 
+    appropriately when loading takes place.
+    """
+
+    def _execute(self, project_name: str):
```

Comment:
Ideally each `TestCase` should verify its result to some extent - you can argue that this is a success if none of the steps raises an exception, but it would be more meaningful to check e.g. if there's at least one instance in the list afterwards, wdyt?

Reply:
Even more appropriate, can we check if all instances listed are part of the specified project? Not sure if we display the project in the table right know TBH...

Reply:
Regarding the second idea. No, we do not display the project in the table, or elsewhere and its also not part of the instance object in the code (which would theoretically be possible to add). 

Regarding the first point. I am contemplating right now whether to include the "needs to have at least one instance". There is no guarantee at all that a project has an instance, we only know that the default project has one, since it is reserved. If I do that, I would rename the Test to `SelectProjectAndVerifyItHasAnInstance`.